mr/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandlerBaseTest.java

```diff
@@ -176,6 +177,40 @@ public void testScanEmptyTable() throws IOException {
     Assert.assertEquals(0, rows.size());
   }
 
+  @Test
+  public void testDecimalTableWithPredicateLiterals() throws IOException {
+    shell.setHiveSessionValue("hive.vectorized.execution.enabled", "false");
+    Schema schema = new Schema(required(1, "decimal_field", Types.DecimalType.of(7, 2)));
+    List<Record> records = TestHelper.RecordsBuilder.newInstance(schema)
+            .add(BigDecimal.valueOf(8500, 2)) // 85.00
+            .add(BigDecimal.valueOf(100.56))
+            .add(BigDecimal.valueOf(100.57))
```

Comment:
I think test values should be created using strings so we know that we know the exact BigDecimal values. Converting from float seems unreliable. The one that uses an integer and a specific scale seems fine.

Reply:
makes sense, using strings now